code/modules/shuttle/emergency.dm

```diff
@@ -221,6 +240,44 @@
 
 	priority_announce("The emergency shuttle has been called. [redAlert ? "Red Alert state confirmed: Dispatching priority shuttle. " : "" ]It will arrive in [timeLeft(600)] minutes.[reason][SSshuttle.emergencyLastCallLoc ? "\n\nCall signal traced. Results can be viewed on any communications console." : "" ]", null, 'sound/AI/shuttlecalled.ogg', "Priority")
 
+/obj/docking_port/mobile/emergency/proc/legitimize_call() //Checks are sorted in rough order of processing cost
+	if(SSshuttle.force_shuttle) //Set manually by certain "Oh shit" events, like narsie.
+		message_admins("Shuttle will arrive due to you know what.")
+		return 1
+
+	if(world.time >= (config.shuttle_boredom_check * 600)) //Extended mercy and/or boring/ineffective antags
+		message_admins("Shuttle will arrive due to long round length.")
+		return 1
+
+	var/list/living_crew_bodies = list()
+	var/list/living_crew_minds	= list()
+	for(var/mob/Player in mob_list)
+		if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player))
+			living_crew_bodies	+= Player
+			living_crew_minds	+= Player.mind
+
+	if(living_crew_bodies.len / joined_player_list.len <= config.shuttle_life_check) //Dead people everywhere
+		message_admins("Shuttle will arrive due to rampant death.")
+		return 1
+
+	var/list/antagonist_crew_minds	= list()
+	for(var/datum/mind/M in living_crew_minds)
+		if(M.special_role)
+			antagonist_crew_minds += M
+
+	if(antagonist_crew_minds.len / living_crew_minds.len >= config.shuttle_antag_overrun) //Station ruled by antags
+		message_admins("Shuttle will arrive due to antagonist infestation.")
+		return 1
+
+	var/datum/station_state/current_state = new /datum/station_state()
+	current_state.count()
+	if((start_state.score(current_state)) < config.shuttle_infrastructure_check) //Station bombed to hell/singulo'd
+		message_admins("Shuttle will arrive due to badly damaged station.")
+		return 1
+
+	message_admins("Shuttle will recall automatically. If you feel the shuttle should come, <A HREF='?_src_=holder;force_shuttle=\ref[usr]'>click here</A>.")
```

Comment:
The autocommand report has a message from a "rear-admiral", the admins should have the button to replace the message if they want.